CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -455,4 +455,23 @@ public static function getAllDeferredFinancialAccount() {
     return $financialAccount;
   }
 
+  /**
+   * Get Organization Name associated with Financial Account.
+   *
+   * @return array
+   *
+   */
+  public static function getOrganizationNames() {
+    $result = civicrm_api3('FinancialAccount', 'get', array(
+      'return' => array("contact_id.organization_name", "contact_id"),
+      'contact_id.is_deleted' => 0,
+      'options' => array('limit' => 0),
+    ));
+    $organizationNames = array();
+    foreach ($result['values'] as $values) {
+      $organizationNames[$values['contact_id']] = $values['contact_id.organization_name'];
+    }
+    return $organizationNames;
+  }
```

Comment:
I think this function ought to take a `$checkPermissions` param which defaults to `TRUE`, and passes that into the api call.